hypha/apply/funds/tests/test_views.py

```diff
@@ -207,6 +207,30 @@ def test_can_create_project(self):
         self.assertTrue(hasattr(submission, 'project'))
         self.assertEquals(submission.project.id, project.id)
 
+    def test_screen_application_primary_action(self):
+        def assert_screen_application(content, action_displayed):
+            screen_application_button = '<a data-fancybox data-src="#screen-application" class="button button--bottom-space button--primary button--full-width is-not-disabled" href="#">Screen application</a>'
+
+            sidebar = content[content.find('<aside class="sidebar'):].replace('\\n', '')
```

Comment:
We have beautifulsoup installed on this project - going to be easier to use it long term. Use that to extract the sidebar and get all anchor elements from within that. You can then confirm that it contains the correct number of "Screen Application" references. 

Copying in the entire HTML fragment feels like it's going to be a pain to maintain long term. 

maybe something like:
```python
buttons = BeautifulSoup(response.content, 'html5lib').find(class_='sidebar').find_all('a', text='Screen applications')
self.assertEqual(len(buttons), 1)
```

Reply:
yes, that makes more sense. I've updated the tests with this